File: fraudshield/android/app/src/main/java/com/citadel/fraudshield/v2/ScanWorker.java
package com.citadel.fraudshield.v2;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import java.util.List;
import java.util.Map;

/**
 * ScanWorker performs periodic background scans.
 * Converted from Kotlin CoroutineWorker to Java Worker to bypass K2 compiler bug.
 */
public class ScanWorker extends Worker {

    public ScanWorker(@NonNull Context appContext, @NonNull WorkerParameters workerParams) {
        super(appContext, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        Log.d("ScanWorker", "Starting periodic background scan");

        try {
            Map<String, Object> results = ScannerEngine.performFullScan(getApplicationContext());
            Object riskyApps = results.get("riskyApps");
            int riskyCount = 0;
            if (riskyApps instanceof List) {
                riskyCount = ((List<?>) riskyApps).size();
            }
            Log.d("ScanWorker", "Background scan completed. Found " + riskyCount + " risky apps.");
            return Result.success();
        } catch (Exception e) {
            Log.e("ScanWorker", "Background scan failed", e);
            return Result.retry();
        }
    }
}

File: fraudshield/android/app/src/main/kotlin/com/citadel/fraudshield/v2/ScanWorker.kt
package com.citadel.fraudshield.v2

import android.content.Context
import androidx.work.CoroutineWorker
import androidx.work.WorkerParameters
import android.util.Log

class ScanWorker(appContext: Context, workerParams: WorkerParameters) :
    CoroutineWorker(appContext, workerParams) {

    override suspend fun doWork(): Result {
        Log.d("ScanWorker", "Starting periodic background scan")
        
        return try {
            // Phase 3: Background scanning using optimized engine
            val results = ScannerEngine.performFullScan(applicationContext)
            Log.d("ScanWorker", "Background scan completed. Found ${ (results["riskyApps"] as? List<*>)?.size ?: 0 } risky apps.")
            
            // Note: In Phase 4, we would send this to the backend (via ApiService equivalent in native) 
            // or trigger a system notification if critical threats are found.
            
            Result.success()
        } catch (e: Exception) {
            Log.e("ScanWorker", "Background scan failed", e)
            Result.retry()
        }
    }
}
